src/stringProcessing/relevantWords.js

```diff
@@ -1,312 +1,188 @@
-		const combination = wordCombination.getCombination();
+import getWords from "../stringProcessing/getWords";
+import { normalizeSingle } from "../stringProcessing/quotes";
+import WordCombination from "../values/WordCombination";
+import functionWordListsFactory from "../helpers/getFunctionWords";
+import getStemForLanguageFactory from "../helpers/getStemForLanguage";
 
-		if ( ! has( occurrences, combination ) ) {
-			occurrences[ combination ] = wordCombination;
-		}
-
-		occurrences[ combination ].incrementOccurrences();
-	} );
-
-	return values( occurrences );
-}
+const functionWordLists = functionWordListsFactory();
+const stemFunctions = getStemForLanguageFactory();
+const digitsAndPunctuation = /[1234567890‘’“”"'.…?!:;,¿¡«»&*@#±^%$|~=+§`[\](){}⟨⟩<>/\\–\-\u2014\u00d7\s]/g;
 
 /**
- * Returns only the relevant combinations from a list of word combinations. Assumes
- * occurrences have already been calculated.
+ * Returns only the relevant combinations from a list of word combinations.
  *
  * @param {WordCombination[]} wordCombinations A list of word combinations.
+ *
  * @returns {WordCombination[]} Only relevant word combinations.
  */
 function getRelevantCombinations( wordCombinations ) {
 	wordCombinations = wordCombinations.filter( function( combination ) {
-		return combination.getOccurrences() !== 1 &&
-			combination.getRelevance() !== 0;
+		return ( combination.getOccurrences() !== 1 && combination.getWord().replace( digitsAndPunctuation, "" ) !== "" );
 	} );
 	return wordCombinations;
 }
 
 /**
- * Sorts combinations based on their relevance and length.
+ * Sorts word combinations based on their number of occurrences and length.
  *
  * @param {WordCombination[]} wordCombinations The combinations to sort.
+ *
  * @returns {void}
  */
 function sortCombinations( wordCombinations ) {
 	wordCombinations.sort( function( combinationA, combinationB ) {
-		const difference = combinationB.getRelevance() - combinationA.getRelevance();
-		// The combination with the highest relevance comes first.
+		const difference = combinationB.getOccurrences() - combinationA.getOccurrences();
+		// The combination with the highest number of occurrences comes first.
 		if ( difference !== 0 ) {
 			return difference;
 		}
-		// In case of a tie on relevance, the longest combination comes first.
-		return combinationB.getLength() - combinationA.getLength();
-	} );
-}
 
-/**
- * Filters word combinations that consist of a single one-character word.
- *
- * @param {WordCombination[]} wordCombinations The word combinations to filter.
- * @returns {WordCombination[]} Filtered word combinations.
- */
-function filterOneCharacterWordCombinations( wordCombinations ) {
-	return wordCombinations.filter( function( combination ) {
-		return ! ( combination.getLength() === 1 && combination.getWords()[ 0 ].length <= 1 );
-	} );
-}
+		// In case of a tie on occurrence number, the alphabetically first combination comes first.
+		const stemA = combinationA.getStem();
```

Comment:
Do we want to use [localeCompare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) here?

```js
	return localeCompare( combinationA.getStem(), combinationB.getStem() );
```

Edit:
Looking beyond this I see you use the same thing in `collapseRelevantWordsOnStem`.

Reply:
Nice one! Improves the core dramatically 👍